test/ElasticSearchIndexerTest.php

```diff
@@ -88,6 +90,107 @@ public function testGetIndexNameByDocumentForUnclassifieds(): void
         $this->assertSame(IndexerInterface::UNCLASSIFIEDS_DOCUMENTS_INDEX, $indexName);
     }
 
+    public function testBuildIndexBulkErrorResponseAfter100(): void
+    {
+        $this->expectException(ClientErrorResponseException::class);
+        $this->expectExceptionMessage('some reason; some other reason');
+
+        $document = $this->createMock(IndexDocument::class);
+        $document->expects($this->any())
+            ->method('getBody')
+            ->willReturn([
+                'type' => [
+                    TaoOntology::CLASS_URI_ITEM,
+                ],
+            ]);
+
+        $document->expects($this->any())
+            ->method('getId')
+            ->willReturn('some_id');
+
+        $this->client
+            ->expects($this->atMost(100))
+            ->method('bulk')
+            ->willReturn([
+                'errors' => true,
+                'items' => [
+                    [
+                        [
+                            'error' => [
+                                'reason' => 'some reason',
+                            ],
+                        ],
+                    ],
+                    [
+                        [
+                            'error' => [
+                                'reason' => 'some other reason',
+                            ],
+                        ],
+                    ],
+                ],
+            ]);
+
+        $iterator = $this->createIterator($this->getMultipleDocs($document, 101));
+
+
+        $this->sut->buildIndex($iterator);
+    }
+
+    private function getMultipleDocs(MockObject $doc, $quantity)
```

Comment:
I approved the PR, but we could avoid that by making the `INDEXING_BLOCK_SIZE` injectable / customizable, so yuo do not need to iterate 100 results to test. I think this test will be slow at some point if we do not do it.

Thanks!

Reply:
Can you please move the private method as the last? It is in our CR guideline